### ISSUE TITLE
FileDownload fails with exception when file not available

### ISSUE THREAD

**Issue by aj062** [buildbot/buildbot#5877]
transfer.FileDownload fails with exception when expected file is not available. It should be gracefully handled, and the step should just fail, instead of causing an exception.

Exception:
```Traceback (most recent call last):
  File "/usr/local/lib64/python3.8/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/usr/local/lib64/python3.8/site-packages/twisted/internet/defer.py", line 1475, in gotResult
    _inlineCallbacks(r, g, status)
  File "/usr/local/lib64/python3.8/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/usr/local/lib64/python3.8/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
--- <exception caught here> ---
  File "/usr/local/lib/python3.8/site-packages/buildbot/process/buildstep.py", line 628, in startStep
    self.results = yield self.run()
  File "/usr/local/lib64/python3.8/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/usr/local/lib/python3.8/site-packages/buildbot/steps/transfer.py", line 459, in run
    fp = open(source, 'rb')
builtins.FileNotFoundError: [Errno 2] No such file or directory: '123456.zip'
```

**Comment by aj062** [comment]
Seems like the fix would be to add FileNotFoundError (along-with IOError) while handling the exception at: https://github.com/buildbot/buildbot/blob/master/master/buildbot/steps/transfer.py#L460

**Comment by tardyp** [comment]
I think this is the intended behavior:

There are tests ensuring the output is exception
https://github.com/buildbot/buildbot/blob/2b70424f8960b5b0d480589e9d711dea40556562/master/buildbot/test/unit/steps/test_transfer.py#L454-L454

Whether this is suitable I guess this is arguable.

**Comment by aj062** [comment]
That test seems to be for uploading, maybe you wanted to comment on https://github.com/buildbot/buildbot/issues/5878 instead.

Since there is an explicit try/except block at https://github.com/buildbot/buildbot/blame/master/master/buildbot/steps/transfer.py#L460 with log message saying "File {!r} not available at master", that indicate that the expected behavior is to handle the exception gracefully.

**Comment by pocek** [comment]
I think this got fixed by #5959.

**Comment by p12tic** [comment]
Indeed, this bug can be closed.
